File: ChickInRun/Assets/Script/Goal.js
#pragma strict
private var goalNum : int;
public var gate : GameObject;


function Start () {

}

function Update () {

	goalNum = gate.gameObject.GetComponent(GateOpen).openNum;

}

function OnTriggerEnter (col : Collider){

	if(col.gameObject.tag == "Chick"){
		if(goalNum == 12){
			yield WaitForSeconds(1.0f);
			Application.LoadLevel("Ending");
		}
	}
}

File: ChickInRun/Assets/Script/Goal.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.SceneManagement;

public class Goal : MonoBehaviour
{
    private int goalNum;
    public GameObject gate;

    private void OnTriggerEnter(Collider other)
    {
        if(other.gameObject.tag == "Chick")
        {
            if(goalNum==12)
            {
                StartCoroutine("WaitGoal");
                SceneManager.LoadScene("Ending");
            }
        }
    }
    private IEnumerator WaitGoal()
    {
        yield return new WaitForSeconds(1.0f);
    }


    // Start is called before the first frame update
    void Start()
    {
        
    }

    // Update is called once per frame
    void Update()
    {
        goalNum = gate.gameObject.GetComponent<GateOpen>().openNum;
    }
}
